src/compiler/scala/tools/nsc/typechecker/TypeDiagnostics.scala

```diff
@@ -544,6 +545,8 @@ trait TypeDiagnostics {
               case _                 =>
             }
           }
+          val expandee = macroExpandee(t)
+          if (expandee ne t) traverse(expandee)
           super.traverse(t)
```

Comment:
Does this mean: if this is a macro and t is not the original expandee, then lint it; always lint the expanded. The flag is checked in traverse before actually descending into either tree. So it's either both trees or neither. Is that it?